lib/types/src/batch_signature.rs

```diff
@@ -0,0 +1,63 @@
+use alloy::primitives::{Address, Signature as AlloySignature, SignatureError};
+use alloy::signers::Signer;
+use alloy::signers::local::PrivateKeySigner;
+use alloy::sol_types::SolValue;
+use serde::{Deserialize, Serialize};
+use zksync_os_contract_interface::IExecutor::CommitBatchInfoZKsyncOS;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct BatchSignatureSet(Vec<BatchSignature>);
```

Comment:
Similarly, `types` crate should only contains basic types and this is a lot more domain-specific.

I propose you move this and maybe some other batcher-related types into a separate crate (can be done separately ofc). And name it "batcher-types" or something like that to signify that these are types shared by the entire batcher subsystem.

There is an argument for moving `BlockMerkleTreeData` there too